### DISCUSSION TITLE
Apollo and createNamed() client

### DISCUSSION THREAD

**Discussion by danieler1981** [dotansimha/graphql-code-generator#4972] [General] [answered]
Hi,

I'm trying to setup a project based on Angular and ApolloClient and I'm having issue with Named-Client on Apollo and Codegen.

Our context:
The project use default Apollo Client and also "Named Client": in this way we can use default-client for query/mutation and the named-client for handle file-upload mutation.

Happy scenario:
Codegen is working fine with default-apollo-client and is able to handle well all query/mutations.
We construct the SDK like this `constructor(apollo: Apollo.Apollo) {` and we pass an instance of "this.apollo" from our code.

Problem:
For file upload we consume the mutation with named-client like this: `this.apollo.use('withUpload').mutate()` and this is working in our code but isn't with CodeGen. From my understanding seems like codegen expect to receive an instance of Apollo `constructor(apollo: Apollo.Apollo) {`but if we pass `this.apollo.use('withUpload')` this code with not work because use() will return an instance of ApolloBase.

Any suggestions to solve this scenario?

**Comment by danieler1981** [comment, marked as answer]
Solved by implementing GraphQL Directive `@namedClient`.

**Reply by Brokkolii** [reply]
Hello, im struggling with the exact same problem. Can you please provide context on your solution?
